plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/QueryBuilder.java

```diff
@@ -143,6 +143,23 @@ else if (baseRelation instanceof JdbcQueryRelationHandle) {
         return new PreparedQuery(sql, accumulator.build());
     }
 
+    public PreparedQuery prepareDelete(
+            ConnectorSession session,
+            Connection connection,
+            JdbcRelationHandle baseRelation,
+            TupleDomain<ColumnHandle> tupleDomain)
+    {
+        String sql = "DELETE FROM " + getRelation(((JdbcNamedRelationHandle) baseRelation).getRemoteTableName());
+
+        ImmutableList.Builder<QueryParameter> accumulator = ImmutableList.builder();
+
+        List<String> clauses = toConjuncts(session, connection, tupleDomain, accumulator::add);
+        if (!clauses.isEmpty()) {
+            sql += " WHERE " + Joiner.on(" AND ").join(clauses);
+        }
+        return new PreparedQuery(sql, accumulator.build());
```

Comment:
Can we extract it for both `SELECT` and `DELETE` ?

Reply:
I don't think it is useful.